app/controllers/admin/merchants_controller.rb

```diff
@@ -13,16 +13,22 @@ def edit
 
   def update
     merchant = Merchant.find(params[:id])
-    if params[:merchant][:name]
-      if merchant.update(merchant_params)
-        redirect_to admin_merchant_path(merchant)
-        flash[:success] = "Merchant has been successfully updated"
-      end
+    if params[:status] == "1"
+      merchant.update(status: 1)
+      redirect_back fallback_location: admin_merchant_path
```

Comment:
I have not seen this fallback location before, had to look that up, pretty cool and smart